File: event-bus-rabbit/src/main/java/pegasus/eventbus/amqp/AmqpFactory.java
package pegasus.eventbus.amqp;

import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

import pegasus.eventbus.client.EventBusConnectionParameters;
import pegasus.eventbus.client.EventBusFactory;
import pegasus.eventbus.client.EventManager;

public class AmqpFactory implements EventBusFactory {

    private static final Logger LOG = LoggerFactory.getLogger(AmqpFactory.class);

    public EventManager getEventManager(String clientName, EventBusConnectionParameters connectionParameters) {

        LOG.trace("Getting new instance of EventManager for {}.", clientName);

        // @todo - very ugly cast. should handle this more elegantly
        return new AmqpEventManager(AmqpConfiguration.getDefault(clientName, (AmqpConnectionParameters) connectionParameters));
    }

}

File: event-bus-rabbit/src/main/dotnet/amqp/AmqpFactory.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

using log4net;

using pegasus.eventbus.client;


namespace pegasus.eventbus.amqp
{
    public class AmqpFactory : IEventBusFactory
    {
        private static ILog LOG = LogManager.GetLogger(typeof(AmqpFactory));


        public IEventManager GetEventManager(string clientName, EventBusConnectionParameters connectionParameters)
        {
            LOG.DebugFormat("Getting new instance of IEventManager for {0}", clientName);

            // please, do not think that this doesn't make me want to puke.
            // this is how the java client is implemented, and until we get
            // a chance to refactor, I am just following their lead.
            return new AmqpEventManager(AmqpConfiguration.GetDefault(clientName, (AmqpConnectionParameters)connectionParameters));
        }
    }
}
